client/src/lib/data.ts

```diff
@@ -0,0 +1,24 @@
+type newEventEntry = {
+  tripName: string;
+  startDate: Date;
+  endDate: Date;
+};
+
+export async function addEvent(newEvent: newEventEntry) {
+  try {
+    const reqConfig = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(newEvent),
+    };
+    const res = await fetch('/api/event', reqConfig);
+    if (!res.ok) {
+      throw new Error(`Error status ${res.status}`);
+    }
+    // return await res.json(res);
+  } catch (error) {
+    console.log((error as Error).message);
```

Comment:
Since `error` may not actually be an `Error`, it's best to write this as:
`console.error(error instanceof Error ? error.message : error);`

And having said that, if you are just logging it, you really want to always just log the entire error, so actually it should just be:
`console.error('Fetch error', error);`